src/components/RichTextPopup/RichTextPopup.js

```diff
@@ -107,13 +122,26 @@ const RichTextPopup = () => {
     return format;
   };
 
-  const handleClick = format => () => {
+  const handleTextFormatChange = format => () => {
     const { index, length } = editorRef.current.getSelection();
     const currentFormat = editorRef.current.getFormat(index, length);
 
     applyFormat(format, !currentFormat[format]);
   };
 

```

Comment:
Hmm didn't quite get why we need a `setTimeout` here. Can you explain a bit, and maybe add a comment in the code?

It'd be nice if we can get rid of it, as at a first glance this seems kind of unreliable.

If we need this `setTimeout` to re-position the popup so that the math symbol table doesn't get cut off, then I think a better way is to have another effect. It will look something like this:

```js
useEffect(() => {
 // calculate position
 // set css position
}, [symbolsVisible]);
```

The effect should run after the DOM has been updated. So when it runs we should have the correct height of the popup.